htdocs/xoops_lib/modules/protector/admin/prefix_manager.php

```
@@ -4,18 +4,27 @@
 
 const PREFIX_INVALID_CHAR_PATTERN = '/[^0-9A-Za-z_-]/';
 
+/**
+ * Validate a DB prefix: reject (die) if it contains characters outside [A-Za-z0-9_-].
+ * Returns the validated prefix unchanged.
+ */
+function validatePrefix(string $raw): string
+{
+    if (preg_match(PREFIX_INVALID_CHAR_PATTERN, $raw)) {
+        die('wrong prefix');
```

PHPMD (2.15.0)

[error] The function validatePrefix() contains an exit expression. (undefined)

(ExitExpression)